docs/api/synthetics/monitors/update-monitor-api.asciidoc

```diff
@@ -0,0 +1,271 @@
+
+Updates a new monitor with the specified attributes. The required and default fields may vary based on the monitor type.
+
+=== {api-request-title}
+
+`PUT <kibana host>:<port>/api/synthetics/monitors/<config_id>`
+
+`PUT <kibana host>:<port>/s/<space_id>/api/synthetics/monitors/<config_id>`
+
+=== {api-prereq-title}
+
+You must have `all` privileges for the *Synthetics* feature in the *{observability}* section of the
+<<kibana-feature-privileges,{kib} feature privileges>>.
+
+
+[[update-monitor-api-path-params]]
+=== {api-path-parms-title}
+
+`config_id`::
+(Required, string) The ID of the monitor that you want to update.
+
+`space_id`::
+(Optional, string) An identifier for the space. If `space_id` is not provided in
+the URL, the default space is used.
+
+[[monitor-update-body]]
+==== Request Body
+
+The request body should contain the attributes of the monitor you want to update. The required and default fields differ depending on the monitor type:
+
+*Common Fields*:
+
+`name`:: (Required, string): The monitor's name.
+
+`type`:: (Required, string): The monitor type (e.g., `http` , `tcp` , `icmp` or `browser`) can't be changed after the monitor is created.
+
+`schedule`:: (Optional, number): The monitor's schedule in minutes. Supported values are `1`, `3`, `5`, `10`, `15`, `30`, `60`, `120` and `240`.
+  * The default value is `3` minutes for HTTP, TCP, and ICMP monitors.
+
+  * The default value is `10` minutes for Browser monitors.
+
+:private-locations-url: https://www.elastic.co/guide/en/observability/current/synthetics-private-location.html
+:locations-command: https://www.elastic.co/guide/en/observability/current/synthetics-command-reference.html#elastic-synthetics-locations-command
+
+`locations`:: (https://github.com/elastic/synthetics/blob/main/src/locations/public-locations.ts#L28-L37[`Array<SyntheticsLocationsType>`])
+Where to deploy the monitor. Monitors can be deployed in multiple locations so that you can detect differences in availability and response times across those locations.
++
+To list available locations you can:
++
+  * Run the {locations-command}[`elastic-synthetics locations`] command
+  with the deployment's {kib} URL.
+  * Go to *Synthetics* -> *Management* and click *Create monitor*.
+    Locations will be listed in _Locations_.
+
+`private_locations` (`Array<string>`)::
+
+The {private-locations-url}[Private Locations] to which the monitors will be deployed. These Private Locations refer to locations hosted and managed by you, whereas
+`locations` are hosted by Elastic. You can specify a Private Location using the location's name.
++
+To list available Private Locations you can:
++
+  * Run the {locations-command}[`elastic-synthetics locations`] command
+  with the deployment's {kib} URL.
+  * Go to *Synthetics* -> *Settings* and click *Private locations*.
+  Private Locations will be listed in the table.
+
+[NOTE]
+====
+You may provide `locations` or `private_locations`, or both. At least one is required.
```

Comment:
```suggestion
You may provide `locations` or `private_locations`, or both. At least one is required.
```

This isn't actually true, you can submit this request without specifying anything location-related as we are just updating the existing monitor and it will keep its locations set up.

Reply:
https://github.com/elastic/kibana/pull/169928/commits/81c6928c9bca943e24a338481562609f2025e029